Packs/Tenable_io/Integrations/Tenable_io/Tenable_io.py

```diff
@@ -1380,51 +1664,280 @@ def export_scan_command(args: dict[str, Any], client: Client) -> PollResult:
+        actor_id: fetch audit logs with matching actor id.
+        target_id:fetch audit logs with matching target id.
+        limit: limit number of audit logs to get.
 
-    client = Client(
-        BASE_URL,
-        verify=USE_SSL,
-        proxy=USE_PROXY,
-        ok_codes=(200,),
-        headers=HEADERS
-    )
+    Returns: CommandResults of audit logs from API.
+
+    """
+    audit_logs = client.get_audit_logs_request(from_date=from_date,
+                                               to_date=to_date,
+                                               actor_id=actor_id,
+                                               target_id=target_id,
+                                               limit=limit)
+
+    readable_output = tableToMarkdown('Audit Logs List:', audit_logs,
+                                      removeNull=True,
+                                      headerTransform=string_to_table_header)
+
+    results = CommandResults(readable_output=readable_output,
+                             raw_response=audit_logs)
+    return results, audit_logs
+
+
+''' FETCH COMMANDS '''
+
+
+def fetch_events_command(client: Client, first_fetch: datetime, last_run: dict, limit: int = 1000):
+    """
+    Fetches audit logs.
+    Args:
+        client: Client class object.
+        first_fetch: time to first fetch from.
+        last_run: last run object.
+        limit: number of audit logs to max fetch.
+
+    Returns: vulnerabilities, audit logs and updated last run object
+
+    """
+
+    last_fetch = last_run.get('last_fetch_time')
+    last_index_fetched = last_run.get('index_audit_logs', 0)
+    if not last_fetch:
+        start_date = first_fetch.strftime(DATE_FORMAT)
+    else:
+        start_date = last_fetch  # type: ignore
+
+    audit_logs: List[dict] = []
+    audit_logs_from_api = client.get_audit_logs_request(from_date=start_date)
+
+    if last_index_fetched < len(audit_logs_from_api):
+        audit_logs.extend(audit_logs_from_api[last_index_fetched:last_index_fetched + limit])
+
+    next_run: str = datetime.now(tz=timezone.utc).strftime(DATE_FORMAT)
+    last_run.update({'index_audit_logs': len(audit_logs) + last_index_fetched if audit_logs else last_index_fetched,
+                     'last_fetch_time': next_run})
+    demisto.info(f'Done fetching {len(audit_logs)} audit logs, Setting {last_run=}.')
+    return audit_logs, last_run
+
+
+def fetch_assets_command(client: Client, assets_last_run):
+    """
+    Fetches assets.
+    Args:
+        assets_last_run: last run object.
+        client: Client class object.
+
+    Returns:
+        assets fetched from the API.
+    """
+    assets = []
+    # if already in assets_last_run meaning its still polling chunks from api
+    export_uuid = assets_last_run.get('assets_export_uuid')
+    # if exists, still downloading chunks from prev fetch call
+    available_chunks = assets_last_run.get('assets_available_chunks', [])
+    if available_chunks:
```

Comment:
is it possible to get available chunks and export uuid ?

Reply:
no, because available chunks would return from previous call which is already with finished status, and in this case the export uuid would not be stored in the lastrun